File: backend/app/routers/sitemap.py
"""GET /sitemap/locations — public index of every ACTIVE location page, used by
the frontend's `sitemap.xml` (`frontend/src/app/sitemap.ts`) to list every
location URL with the right canonical form. See docs/API_CONTRACTS.md
"GET /sitemap/locations"."""
from __future__ import annotations

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.dependencies.db import get_db
from app.dependencies.pagination import Pagination, pagination_params
from app.schemas.restaurant import PublicLocationIndexResponse
from app.services import restaurant_service

router = APIRouter(prefix="/sitemap", tags=["sitemap"])


@router.get("/locations", response_model=PublicLocationIndexResponse)
async def list_public_locations(
    pagination: Pagination = Depends(pagination_params),
    db: AsyncSession = Depends(get_db),
) -> PublicLocationIndexResponse:
    return await restaurant_service.list_public_location_index(db, pagination)

File: frontend/src/lib/api/sitemap.ts
// Typed client for GET /sitemap/locations — docs/API_CONTRACTS.md. Public;
// used only by app/sitemap.ts to list every ACTIVE location page.
import { apiFetch, toQueryString } from "./client";
import type { PaginatedResponse, PaginationParams } from "@/types/common";
import type { PublicLocationIndexItem } from "@/types/restaurant";

export async function getPublicLocationIndex(
  params: PaginationParams = {}
): Promise<PaginatedResponse<PublicLocationIndexItem>> {
  const query = toQueryString({ page: params.page, page_size: params.page_size });
  return apiFetch<PaginatedResponse<PublicLocationIndexItem>>(
    `/sitemap/locations${query}`,
    { method: "GET" },
    { revalidateSeconds: 3600 }
  );
}
